be/src/runtime/current_thread.h

```diff
@@ -245,23 +245,15 @@ class CurrentThreadCatchSetter {
         } else {                                                                                                       \
             return Status::MemoryLimitExceeded("Mem usage has exceed the limit of BE");                                \
         }                                                                                                              \
+    }                                                                                                                  \
+    catch (std::runtime_error const& e) {                                                                              \
```

Comment:
Don't all std::runtime_error meaning bad alloc, why we could handle it by this way?

Reply:
Although it could throw `std:: runtime_error` by other reasons, it is OK to try catch it and return an error status to FE. The reason is that, the codes using `TRY_CATCH_BAD_ALLOC` is expcetion-safe, and FE will receive an error status.

`ExprContext::evaluate` also try catch `std:: runtime_error`, when string function generate a large column.